modules/public/api.py

```diff
@@ -53,20 +57,25 @@
 }
 
 
-class InvalidResponseContent(Exception):
-    pass
-
-
-class ApiErrorResponse(Exception):
-    pass
+def process_response(response):
+    try:
+        body = response.json()
+    except ValueError as decode_error:
+        error_message = 'JSON decoding failed: {}'.format(decode_error)
+        raise InvalidResponseContent(error_message)
 
+    if not response.ok:
+        if 'error_list' in body:
+            api_error_exception = ApiErrorResponse("Error list")
```

Comment:
Similarly, since only an "Error list" is going to contain an `errors` property, we should probably be explicit about this by making it its own type: `ApiResponseErrorList`.